datalad/support/annexrepo.py

```diff
@@ -736,57 +736,69 @@ def proxy(self, git_cmd, **kwargs):
                                        annex_options=['--'] + git_cmd,
                                        **kwargs)
 
-    @normalize_path
-    def get_file_key(self, file_):
+    @normalize_paths
+    def get_file_key(self, files):
         """Get key of an annexed file.
 
         Parameters
         ----------
-        file_: str
-            file to look up
+        files: str or list
+            file(s) to look up
 
         Returns
         -------
-        str
-            keys used by git-annex for each of the files
+        str or list
+            keys used by git-annex for each of the files;
+            in case of a list an empty string is returned if there was no key
+            for that file
         """
 
-        cmd_str = 'git annex lookupkey %s' % file_  # have a string for messages
+        if len(files) > 1:
```

Comment:
In all other cases there is a switch option for using batched mode, and it is never decided by the number of files. If I see it right you would get now completely different behavior in how special cases reported back. At least with nothing in docstring